languages/go/exercises/concept/maps/.docs/introduction.md

```diff
@@ -0,0 +1,58 @@
+In go, `map` is a built-in data type that represent a key/value store or hash table. In other programming language, you might know `map` as `dict` or associative array. If you're not familiar with such concept, map you can think `map` like a dictionary, which every word is the `key` and the definition is the `element` of `map`.
+
+> Before we begin, I'd like to point you to [go spec for map][gospec-map] and [go blog for map][goblog-map] to dig further into `map`.
+
+Syntactically, `map` looks like this:
+
+```go
+map[KeyType]ElementType
+```
+
+_`KeyType` must be any [comparable type][gospec-comparable], while `ElementType` can be any valid type in go, which means you can store anything from primitive variable to a slice._
+
+It's important to remember that `map` in go is **unordered**, if you try to loop trough a `map` and print the element, you might surprize yourself seeing that your elements printed in random order (give it a try if you like).
+
+It is also important to know that each key is unique, meaning that assigning the same key twice will overwrite the value of the corresponding key.
+
+`map` is reference type, which means if you pass it around, go won't copy the whole map. Instead what go will do is go copy the pointer of the map, this makes passing map to a function or variable cheap. The value of an uninitialized map is `nil`.
```

Comment:
There is quite a bit of information here. Is there any information that the student does _not_ need to solve the exercise? If so, could you remove this information and only have it in the `after.md` document? See [the documentation](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsintroductionmd).

Reply:
I thought the same thing when i was writing the introductions, but I missed a paragraph there, see 9bc5573 , thank you